beerus_core/src/lightclient/beerus.rs

```diff
@@ -20,34 +26,65 @@ pub enum SyncStatus {
     Synced,
 }
 
+#[derive(Clone, Debug)]
+pub struct NodeData {
+    pub block_number: u64,
+    pub state_root: String,
+    pub payload: BTreeMap<u64, BlockWithTxs>,
+}
+
+impl NodeData {
+    pub fn new() -> Self {
+        NodeData {
+            block_number: 0,
+            state_root: "".to_string(),
+            payload: BTreeMap::new(),
+        }
+    }
+}
+
+impl Default for NodeData {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
 /// Beerus Light Client service.
 pub struct BeerusLightClient {
     /// Global configuration.
     pub config: Config,
     /// Ethereum light client.
-    pub ethereum_lightclient: Box<dyn EthereumLightClient>,
+    pub ethereum_lightclient: Arc<RwLock<Box<dyn EthereumLightClient>>>,
     /// StarkNet light client.
-    pub starknet_lightclient: Box<dyn StarkNetLightClient>,
+    pub starknet_lightclient: Arc<Box<dyn StarkNetLightClient>>,
     /// Sync status.
     pub sync_status: SyncStatus,
     /// StarkNet core ABI.
     pub starknet_core_abi: Abi,
     /// StarkNet core contract address.
     pub starknet_core_contract_address: H160,
+    // TODO: Add Payload data
+    pub node: Arc<RwLock<NodeData>>,
 }
 
 impl BeerusLightClient {
-    /// Create a new Beerus Light Client service.
+    //     /// Create a new Beerus Light Client service.
     pub fn new(
         config: Config,
-        ethereum_lightclient: Box<dyn EthereumLightClient>,
-        starknet_lightclient: Box<dyn StarkNetLightClient>,
+        //TODO: Check if we should just have &str as arguments
```

Comment:
There were a few issues when trying this, specially because of the automock set up, I think this will need a little more exploration, is it ok to create an issue and I work on it separately?

Reply:
yup that sounds good